File: src/commands/developer/uptime.ts
import { TofuCommand } from "@/core/command";
import { DurationUtils } from "@/utils/duration";
import { ErisUtils } from "@/utils/eris";

export const uptimeCommand: TofuCommand = {
    config: {
        name: "uptime",
        description: "Check bot's uptime.",
    },
    invoke: async (tofu) => ({
        message: ErisUtils.robotMessage(
            `I've been up and running for **${DurationUtils.prettyMilliseconds(
                tofu.uptime,
                "human",
            )}**!`,
        ),
    }),
};

File: commands/uptime.go
package commands

import (
	"fmt"
	"time"

	"github.com/disgoorg/disgo/discord"
	"github.com/disgoorg/disgo/events"
	"me.zyrouge.tofu/core"
	"me.zyrouge.tofu/utils"
)

func NewTofuUptimeCommand() core.TofuCommand {
	return core.TofuCommand{
		Config: discord.SlashCommandCreate{
			Name:        "uptime",
			Description: "Check bot's uptime.",
			Contexts: []discord.InteractionContextType{
				discord.InteractionContextTypeGuild,
			},
		},
		Invoke: func(tofu *core.Tofu, event *events.ApplicationCommandInteractionCreate) discord.InteractionResponseData {
			uptime := getTofuUptimeString(tofu)
			return discord.NewMessageCreateBuilder().
				SetContent(utils.RobotMessage(fmt.Sprintf("I've been up and running for **%s**!", uptime))).
				Build()
		},
	}
}

func getTofuUptimeString(tofu *core.Tofu) string {
	return time.Since(tofu.StartedAt).Truncate(time.Second).String()
}
